UnityProject/Assets/Scripts/Input System/InteractionV2/DefaultWillInteract.cs

```diff
@@ -72,7 +72,9 @@ public static bool MouseDrop(MouseDrop interaction, NetworkSide side)
 	/// </summary>
 	public static bool AimApply(AimApply interaction, NetworkSide side)
 	{
-		return Validations.CanInteract(interaction.Performer, side);
+		return  Validations.CanInteract(interaction.Performer, side) &&
```

Comment:
Yeah, the idea seems right. That bool inversion is hard to spot though and it's best to avoid calling GetComponent when it's not required. How about this:
```cs
		if ( !Validations.CanInteract(interaction.Performer, side) )
		{
			return false;
		}

		bool? isNotHidden = !interaction.Performer.Player()?.Script.IsHidden;
		return isNotHidden.GetValueOrDefault( true );
```

Reply:
yea that does look a lot better just pushed another commit, thanks.